src/Xamarin.Android.Build.Tasks/Tasks/GetJavaPlatformJar.cs

```diff
@@ -58,22 +60,39 @@ public override bool RunTask ()
 								platform = target_sdk.Value;
 
 							var min_sdk = uses_sdk.Attribute (androidNs + "minSdkVersion");
-							if (min_sdk != null && (!int.TryParse (min_sdk.Value, out int minSdkVersion) || minSdkVersion < XABuildConfig.NDKMinimumApiAvailable)) {
-								Log.LogWarningForXmlNode (
-										code:             "XA4216",
-										file:             AndroidManifest,
-										node:             min_sdk,
-										message:          Properties.Resources.XA4216_MinSdkVersion,
-										messageArgs:      new object [] { min_sdk?.Value, XABuildConfig.NDKMinimumApiAvailable }
-								);
+							if (min_sdk != null) {
+								// NOTE: $(SupportedOSPlatformVersion) will potentially be a version, but could also be an int
+								if (SupportedOSPlatformVersion.IndexOf ('.') == -1) {
+									SupportedOSPlatformVersion += ".0";
+								}
+								int supportedOsPlatformVersionAsInt = 0;
+								if (Version.TryParse (SupportedOSPlatformVersion, out var version)) {
+									supportedOsPlatformVersionAsInt = version.Major;
+								}
+
+								var failedToParseMinSdk = !int.TryParse (min_sdk.Value, out int minSdkVersion);
+
+								if (failedToParseMinSdk || minSdkVersion != supportedOsPlatformVersionAsInt) {
+									Log.LogCodedError ("XA1037", Properties.Resources.XA1037, min_sdk?.Value, SupportedOSPlatformVersion);
+								}
+
+								if (failedToParseMinSdk || minSdkVersion < XABuildConfig.AndroidMinimumDotNetApiLevel) {
+									Log.LogWarningForXmlNode (
+											code:             "XA4216",
+											file:             AndroidManifest,
+											node:             min_sdk,
+											message:          Properties.Resources.XA4216_MinSdkVersion,
+											messageArgs:      new object [] { min_sdk?.Value, XABuildConfig.AndroidMinimumDotNetApiLevel }
+									);
+								}
 							}
-							if (target_sdk != null && (!int.TryParse (target_sdk.Value, out int targetSdkVersion) || targetSdkVersion < XABuildConfig.NDKMinimumApiAvailable)) {
+							if (target_sdk != null && (!int.TryParse (target_sdk.Value, out int targetSdkVersion) || targetSdkVersion < XABuildConfig.AndroidMinimumDotNetApiLevel)) {
```

Comment:
Similar/different to https://github.com/xamarin/xamarin-android/pull/8026/files#r1199283399 , if `targetSdkVersion` is < `$(AndroidMinimumDotNetApiLevel)`, should this be an error?

This doesn't feel reasonable, and I wonder what Java would do with:

```xml
<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="19" />
```

Maybe this *should* remain a warning?  This just feels bananas to me.

Reply:
Maybe both `XA4216` warnings should be repurposed and converted to errors?

Reply:
It seems that Android studio does not produce a warning or error when `targetSdkVersion` < `minSdkVersion`, so we can probably leave the `targetSdkVersion` condition untouched.

I'll try to merge XA1036 into XA4216_MinSdkVersion however, as keeping that warning is redundant.

Reply:
I've repurposed XA4216, it will continue to warn in the case of targetSdkVersion, but will now produce errors for:

 * SupportedOSPlatformVersion < minimum version supported
 * minSdkVersion < minimum version supported